templates/vue/src/components/modals/UserSettingsModal/AvatarForm.vue

```diff
@@ -0,0 +1,453 @@
+        { value: "LongHairNotTooLong", text: "LongHairNotTooLong" },
+        { value: "LongHairShavedSides", text: "LongHairShavedSides" },
+        { value: "LongHairMiaWallace", text: "LongHairMiaWallace" },
+        { value: "LongHairStraight", text: "LongHairStraight" },
+        { value: "LongHairStraight2", text: "LongHairStraight2" },
+        { value: "LongHairStraightStrand", text: "LongHairStraightStrand" },
+        { value: "ShortHairDreads01", text: "ShortHairDreads01" },
+        { value: "ShortHairDreads02", text: "ShortHairDreads02" },
+        { value: "ShortHairFrizzle", text: "ShortHairFrizzle" },
+        { value: "ShortHairShaggyMullet", text: "ShortHairShaggyMullet" },
+        { value: "ShortHairShortCurly", text: "ShortHairShortCurly" },
+        { value: "ShortHairShortFlat", text: "ShortHairShortFlat" },
+        { value: "ShortHairShortRound", text: "ShortHairShortRound" },
+        { value: "ShortHairShortWaved", text: "ShortHairShortWaved" },
+        { value: "ShortHairSides", text: "ShortHairSides" },
+        { value: "ShortHairTheCaesar", text: "ShortHairTheCaesar" },
+        { value: "ShortHairTheCaesarSidePart", text: "ShortHairTheCaesarSidePart" },
+      ],
+      topColorOptions: [
+        { value: "Black", text: "Black" },
+        { value: "Blue01", text: "Blue01" },
+        { value: "Blue02", text: "Blue02" },
+        { value: "Blue03", text: "Blue03" },
+        { value: "Gray01", text: "Gray01" },
+        { value: "Gray02", text: "Gray02" },
+        { value: "Heather", text: "Heather" },
+        { value: "PastelBlue", text: "PastelBlue" },
+        { value: "PastelGreen", text: "PastelGreen" },
+        { value: "PastelOrange", text: "PastelOrange" },
+        { value: "PastelRed", text: "PastelRed" },
+        { value: "PastelYellow", text: "PastelYellow" },
+        { value: "Pink", text: "Pink" },
+        { value: "Red", text: "Red" },
+        { value: "White", text: "White" },
+      ],
+    }
+  },
+  computed: {
+    ...mapState(["avatar"]),
+  },
+  mounted() {
+    this.setExistingAvatar()
+  },
+  methods: {
+    ...mapActions(["updateAvatar"]),
+    setExistingAvatar() {
+      if (!this.avatar || Object.keys(this.avatar).length === 0) return
+      this.isCircle = this.avatar.isCircle
+      this.circleColor = this.avatar.circleColor
+      this.accessoriesType = this.avatar.accessoriesType
+      this.clotheType = this.avatar.clotheType
+      this.clotheColor = this.avatar.clotheColor
+      this.eyebrowType = this.avatar.eyebrowType
+      this.eyeType = this.avatar.eyeType
+      this.facialHairColor = this.avatar.facialHairColor
+      this.facialHairType = this.avatar.facialHairType
+      this.graphicType = this.avatar.graphicType
+      this.hairColor = this.avatar.hairColor
+      this.mouthType = this.avatar.mouthType
+      this.skinColor = this.avatar.skinColor
+      this.topType = this.avatar.topType
+      this.topColor = this.avatar.topColor
+    },

```

Comment:
I think making these computed properties will make everything cleaner and that way you won't need to set them and then use a function to check them.
something like 
```
isCircle(){
 return this?.avatar.isCircle ? this.avatar.isCircle : true
}
```

Reply:
Can't we just merge in the whole avatar object with `this` using `...` notation so that we don't have to repeat this line by line?